vita3k/mem/src/mem.cpp

```diff
@@ -56,6 +57,10 @@ static void alloc_inner(MemState &state, Address address, size_t page_count, All
     const void *const ret = VirtualAlloc(memory, aligned_size, MEM_COMMIT, PAGE_READWRITE);
     LOG_CRITICAL_IF(!ret, "VirtualAlloc failed: {}", log_hex(GetLastError()));
 #else
+    void *addr = mmap(NULL, aligned_size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE, -1, 0);
```

Comment:
I would like to remove the added code and try again. I think the PROT_READ | PROT_WRITE added lower will make it work eventually

Reply:
This is also allocating new pages. We already reserved some.

Reply:
> This is also allocating new pages. We already reserved some.

I think `void *addr = mmap(memory, aligned_size, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE, -1, 0);` should work too, so that it does not allocate new pages?

Reply:
I am not confident about the behavior of that call again. mprotect should do the job. 

If you do it like that, it's like reserve pages at fixed address, which by somehow works on Mac although we already reserved that region.

Just try to remove the new code I tell you.

Reply:
On my computer, [this](https://github.com/Vita3K/Vita3K/pull/585/files#diff-98fe52b4d9c7ab8753d68aaaa13e765bR89) change does the job.

Reply:
Yep, just change to `PROT_READ | PROT_WRITE` will do the job, I am finally testing the result, and will make a new commit.

Reply:
Ok, have made a new commit.